src/main/java/htsjdk/samtools/reference/AbstractFastaSequenceFile.java

```diff
@@ -127,7 +127,7 @@ protected String getAbsolutePath() {
 
     /** Returns the full path to the reference file. */
     public String toString() {
-        return getAbsolutePath();
+        return path == null ? super.toString() : getAbsolutePath();
```

Comment:
I think adding a field that can be specified at construction time that gives this a name would be useful.  Usually the people creating the input streams will have some named source, it's just not easy to convert it to a `Path`, but the name is still useful for error messages.